services/brig/index/src/Brig/Index/Migrations.hs

```diff
@@ -0,0 +1,141 @@
+module Brig.Index.Migrations (migrate) where
+
+import Brig.Index.Migrations.Types
+import qualified Brig.Index.Migrations.V1_ReIndexForSearchTeamMembers as V1_ReIndexForSearchTeamMembers
+import qualified Brig.Index.Options as Opts
+import qualified Cassandra as C
+import qualified Cassandra.Settings as C
+import Control.Lens ((^.), view)
+import Control.Monad.Catch (Exception, MonadThrow, finally, throwM)
+import Data.Aeson ((.=), Value, object)
+import qualified Data.Metrics as Metrics
+import qualified Data.Text as Text
+import qualified Database.V5.Bloodhound as ES
+import Imports
+import qualified Network.HTTP.Client as HTTP
+import System.Logger.Class (Logger)
+import qualified System.Logger.Class as Log
+
+migrate :: Logger -> Opts.ElasticSettings -> Opts.CassandraSettings -> IO ()
+migrate l es cas = do
+  env <- mkEnv l es cas
+  finally (go env) (cleanup env)
+  where
+    go env =
+      runMigrationAction env $ do
+        failIfIndexAbsent (es ^. Opts.esIndex)
+        createMigrationsIndexIfNotPresent
+        runMigrations [V1_ReIndexForSearchTeamMembers.migration]
```

Comment:
Just want to make sure I understand how it works: if you want to add a new migration, you simply add it here to the list.

Then the `runMigrations` function simply checks the latest version and runs the required migrations, right?

Reply:
Yes, this is correct.